File: Session_2/Bài 2.4 Selection Short/Selection_Short.cpp
//đây là chương trình C++ dùng thuật toán selection short

#include <iostream>
#include <vector>

using namespace std;

void Selection_Short (vector <int> &arr)
{
    int n = arr.size();
    for (int i = 0 ; i < n - 1; i++)
    {
        int min_index = i;
        for (int j = i + 1; j < n; j++)
        {
            if (arr[j] < arr[min_index])
            {
                min_index = j;
            }
        }
        if (min_index != i)
        {
            swap(arr[i], arr[min_index]);
        }
    }
}
int main()
{
    vector<int> arr = {5, 2, 9, 1, 6};
    Selection_Short(arr);

    cout << "Ket qua sau khi sap xep: ";
    for (int x : arr)
    {
        cout << x << " ";
    }
    cout << endl;

    return 0;
}

File: Session_2/Bài 2.4 Selection Short/Selection_Short.jl
# Đây là hàm thực hiện thuật toán Selection Sort để sắp xếp mảng
function Selection_Short!(arr)
    n = length(arr)  # Lấy độ dài của mảng
    for i in 1:n-1  # Duyệt qua từng phần tử trong mảng (lặp từ 1 đến n-1)
        min_index = i  # Giả sử phần tử tại vị trí i là nhỏ nhất

        # Tìm phần tử nhỏ nhất trong đoạn còn lại của mảng
        for j in i+1:n
            if arr[j] < arr[min_index]  # So sánh để tìm chỉ số phần tử nhỏ nhất
                min_index = j  # Cập nhật chỉ số của phần tử nhỏ nhất
            end
        end

        # Hoán đổi phần tử nhỏ nhất với phần tử tại vị trí i
        arr[i], arr[min_index] = arr[min_index], arr[i]
    end
end

# Phần chính của chương trình
arr = [5, 2, 9, 1, 5, 6]  # Khởi tạo mảng cần sắp xếp

# Gọi hàm Selection Sort để sắp xếp mảng
Selection_Short!(arr)

# In kết quả sau khi sắp xếp
println("Kết quả Selection Sort: ", arr)  # Output: [1, 2, 5, 5, 6, 9]
